lib/checkPullRequestJob.js

```diff
@@ -0,0 +1,266 @@
+const { SERVER_JOBS_ADMIN } = require('../userWhitelist.json');
+const REGISTRY_FILENAME = 'jobs_registry.py';
+const JOB_FILE_FIX = '_jobs_';
+const CRITICAL_LABEL = 'critical';
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem}
+ */
+const isJobFile = ({ filename }) => {
+  return filename.includes(JOB_FILE_FIX);
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem}
+ */
+const isNewFile = ({ status }) => {
+  return status === 'added';
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const getNewJobsFromFile = (file) => {
+  // This represents an new line added from the git diff.
+  // This is done because classes can only be created in a new line.
+  // '+' represents an addition while '-' represents a deletion.
+  const newLineAdded = '\n+';
+  const changesArray = file.patch.split(newLineAdded);
+  const classIdentifier = 'class';
+  const oneOffJobIdentifier = 'OneOffJob';
+
+  // A new job class should start with class and include OneOffJob in its name.
+  const newJobDefinitions = changesArray.filter(
+    (change) =>
+      change.startsWith(classIdentifier) && change.includes(oneOffJobIdentifier)
```

Comment:
Let us have a regex instead of this and grab the name using a match group. Would be more readable.

Reply:
Done.
Changing to regex helped reveal a bug so it was a good call.
Thanks.